### DISCUSSION TITLE
ESP32 Audio Kit Keys not usable?

### DISCUSSION THREAD

**Discussion by baesek** [marcel-licence/esp32_midi_sampler#77] [Q&A]
Hello,
I tried to just use the buttons as they are without any modifications on my board (ESP32 Audio Kit V2.2 2957 ES8388) 
but there are several issues:
1. if i set the dip switches to (1) OFF, ON,ON,OFF,OFF (5) Key2 (Pin 13) is not working anymore. It only works (is readable) if I set it to ON,ON,ON,OFF,OFF. Is there any documentation on these settings?
2. If i set KEY6 (Pin 5) as INPUT_PULLUP using the pinMode() function audio output is not working anymore.

I would like to use all 6 buttons. Is it really just a design fault that they are not usable as is, or is there a way to fix the issues?

best,
Paul

**Comment by marcel-licence** [comment]
Hi Paul,

for this reason I made the "analog mod" because in the design the GPIO's are shared with multiple functions. I think it is not possible to use all keys and also all other functions at the same time.
I found only the schematic which shows the electrical connections of the different GPIO's. 

Best regards,
Marcel
